### PR TITLE
feat: implement version endpoint

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**
#240

### CODE REVIEW

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-290.nightly-test.deltixhub.io
>E2E tests status: success

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,5 @@
 build/
 bin/
 /data/
-*.log
\ No newline at end of file
+*.log
+src/main/resources/version
```
Comment (astsiapanay) [inline .gitignore:11]: Please add a comment why we don't check-in the resource

File: build.gradle [hunk 2]
```diff
@@ -74,6 +74,16 @@ application {
     mainClass = 'com.epam.aidial.core.AiDial'
 }
 
+distributions {
+    // write current version to the version file
+    beforeEvaluate {
+        def versionFile = file("$projectDir/src/main/resources/version")
+        versionFile.getParentFile().mkdirs()
```
Comment (astsiapanay) [inline build.gradle:81]: what happens if resources or file already exists?

File: .gitignore (hunk 1 above)
Comment (Maxim-Gadalov) [reply .gitignore:11]: added

File: build.gradle (hunk 2 above)
Comment (Maxim-Gadalov) [reply build.gradle:81]: according to the documentation, file and parent directory(s) will be created if not exists

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**